cloudevents-sdk-warp/src/lib.rs

```diff
@@ -0,0 +1,254 @@
+    ///         // do something with the event
+    ///     }
+    ///     );
+    /// ```
+    ///
+    pub fn to_event() -> impl Filter<Extract = (Event,), Error = Rejection> + Copy {
+        warp::header::headers_cloned()
+            .and(warp::body::bytes())
+            .and_then(create_event)
+    }
+
+    async fn create_event(headers: HeaderMap, body: bytes::Bytes) -> Result<Event, Rejection> {
+        match request_to_event(&headers, body) {
+            Ok(event) => Ok(event),
+            Err(e) => Err(warp::reject::custom(CEFilterError {
+                message: e.to_string(),
+            })),
+        }
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use super::to_event;
+        use url::Url;
+        use warp::test;
+
+        use chrono::Utc;
+        use cloudevents::{EventBuilder, EventBuilderV10};
+        use serde_json::json;
+        use std::str::FromStr;
+
+        #[tokio::test]
+        async fn test_request() {
+            let time = Utc::now();
+            let expected = EventBuilderV10::new()
+                .id("0001")
+                .ty("example.test")
+                .source("http://localhost/")
+                .time(time)
+                .extension("someint", "10")
+                .build()
+                .unwrap();
+
+            let result = test::request()
+                .method("POST")
+                .header("ce-specversion", "1.0")
+                .header("ce-id", "0001")
+                .header("ce-type", "example.test")
+                .header("ce-source", "http://localhost/")
+                .header("ce-someint", "10")
+                .header("ce-time", time.to_rfc3339())
+                .filter(&to_event())
+                .await
+                .unwrap();
+
+            assert_eq!(expected, result);
+        }
+
+        #[tokio::test]
+        async fn test_request_with_full_data() {
+            let time = Utc::now();
+            let j = json!({"hello": "world"});
+
+            let expected = EventBuilderV10::new()
+                .id("0001")
+                .ty("example.test")
+                .source(Url::from_str("http://localhost").unwrap())
+                .time(time)
+                .data("application/json", j.to_string().into_bytes())
+                .extension("someint", "10")
+                .build()
+                .unwrap();
+
+            let result = test::request()
+                .method("POST")
+                .header("ce-specversion", "1.0")
+                .header("ce-id", "0001")
+                .header("ce-type", "example.test")
+                .header("ce-source", "http://localhost")
+                .header("ce-someint", "10")
+                .header("ce-time", time.to_rfc3339())
+                .header("content-type", "application/json")
+                .json(&j)
+                .filter(&to_event())
+                .await
+                .unwrap();
+
+            assert_eq!(expected, result);
+        }
```

Comment:
Add another test where you start a server with the filter, then send a bad request, like `ce-specversion: 0.1` and then check if the response contains a 400